xsuite/src/proxy/csproxy.ts

```diff
@@ -0,0 +1,171 @@
+import { addressToBechAddress } from '../data/address';
+import { BroadTx, Proxy, unrawTxRes } from './proxy';
+import { e, eCodeMetadata, EncodableAccount } from '../data/encoding';
+import { Kvs } from '../data/kvs';
+
+export class CSProxy extends Proxy {
+  stopChainSimulator: () => void;
+  autoGenerateBlocks: boolean;
+  verbose: boolean;
+
+  constructor(baseUrl: string, stopChainSimulator: () => void, autoGenerateBlocks: boolean = true, verbose: boolean = false) {
+    super(baseUrl);
+
+    this.stopChainSimulator = stopChainSimulator;
+    this.autoGenerateBlocks = autoGenerateBlocks;
+    this.verbose = verbose;
+  }
+
+  static async setAccount(baseUrl: string, account: EncodableAccount, autoGenerateBlocks: boolean = true, verbose: boolean = false) {
+    const [previousAccount, previousKvs] = await Promise.all([
+      CSProxy.getAccount(baseUrl, account.address),
+      CSProxy.getAccountKvs(baseUrl, account.address),
+    ]);
+    const newAccount = accountToRawAccount(account, previousAccount as any, previousKvs as Kvs);
+
+    if (verbose) {

```

Comment:
The `verbose` was added in order to more easily debug some Chain Simulator issues. I think it is useful to have since the Chain Simulator is not always that reliable, and having this to display extra information is useful.

We could change to use `NODE_DEBUG=http` instead and not have this option, I will revisit this.

Reply:
Let's try `NODE_DEBUG=http` and see if it does the job. If it does, let's remove the `verbose` option.

If it doesn't do the job, the `verbose` option should be introduced as an option of `Proxy` directly, and should make `fetch` to log the call.